File: lib-segment/src/main/java/websiteschema/mpsegment/graph/IGraph.java
package websiteschema.mpsegment.graph;

import websiteschema.mpsegment.dict.IWord;

public interface IGraph {

    public void addVertex();

    public void addEdge(int head, int tail, int weight, IWord obj);

    public int getEdgeWeight(int head, int tail);

    public IWord getEdgeObject(int head, int tail);

    public int[] getAdjacentVertices(int vertex);

    public int getStopVertex(int i, int j);

    public void clear();
}

File: lib-segment-scala/src/main/scala/websiteschema/mpsegment/graph/IGraph.scala
package websiteschema.mpsegment.graph;

import websiteschema.mpsegment.dict.IWord;

trait IGraph {

    def addVertex()

    def addEdge(head: Int, tail: Int, weight: Int, obj: IWord)

    def getEdgeWeight(head: Int, tail: Int) : Int

    def getEdgeObject(head: Int, tail: Int) : IWord

    def getAdjacentVertices(vertex: Int) : Array[Int]

    def getStopVertex(i: Int, j: Int) : Int

    def clear()
}
